### DISCUSSION TITLE
PHY JL1101

### DISCUSSION THREAD

**Discussion by whcir911** [espressif/arduino-esp32#8398] [Q&A]
How can i get the PHY JL1101 driver added to the ESP32arduino core? The maker of the board i have has provided a driver at this link 
"https://github.com/dtlzp/relay_dev_demo/blob/main/esp-idf-dt-patch/components/esp_eth/src/esp_eth_phy_jl1101.c". I'm not sure if this is the correct place to ask this question?

Thanks,
Rick

**Comment by mrengineer7777** [comment]
Hi Rick.  This repository is based on IDF.  The change will have to be be merged at the IDF level, and then this repository will have to expose the new option.

https://github.com/espressif/esp-idf/tree/master/components/esp_eth/src

This repository is currently on IDF branch 4.4, but will be moving to branch 5.1 soon.
https://github.com/espressif/esp-idf/tree/release/v5.1/components/esp_eth/src

Check with IDF team to see which branch you should PR against.

**Comment by Jason2866** [comment]
@whcir911  You can use Tasmota Arduino fork, which has support for PHY JL1101.
Tasmota Arduino esp32 is designed to be used with Platformio.
It is here https://github.com/tasmota/platform-espressif32

**Comment by elvis-epx** [comment]
I had success by copying the driver from ESPHome project (just removing the outermost #ifdef) into ESP-IDF source tree. (Patch attached.) In my case my interest was to make a DTWonder LAN port work on MicroPython (patch for mpy also attached, FWIW). The pins configuration was also found in an ESPHome config file. Translated to mpy:

`lan = network.LAN(mdc=Pin(23), mdio=Pin(18), power=Pin(0), phy_addr=1, phy_type=network.PHY_JL1101, ref_clk=Pin(17), ref_clk_mode=Pin.OUT)
`
[JL1101.zip](https://github.com/espressif/arduino-esp32/files/15392777/JL1101.zip)
[JL1101-mpy.zip](https://github.com/espressif/arduino-esp32/files/15392778/JL1101-mpy.zip)

**Comment by Jason2866** [comment]
The ESPHome driver is an old driver. Wouldn't use with IDF 5.1. Actual driver https://github.com/espressif/esp-idf/pull/13572

**Comment by elvis-epx** [comment]
Looks like your driver uses esp_eth_phy_802_3_autonego_ctrl() which appeared in ESP-IDF 5.1.4 and I have to stick with 5.1.2 since MicroPython is compatible with it. Will try again in the future again when mpy accepts a newer version of ESP-IDF.
